src/app/infobox/infobox.component.ts

```diff
@@ -0,0 +1,42 @@
+import {Component, OnInit, ChangeDetectorRef} from '@angular/core';
+import {Router, ActivatedRoute} from '@angular/router';
+import {Store} from '@ngrx/store';
+import * as fromRoot from '../reducers';
+import {KnowledgeapiService} from '../services/knowledgeapi.service';
+
+@Component({
+  selector: 'app-infobox',
+  templateUrl: './infobox.component.html',
+  styleUrls: ['./infobox.component.css']
+})
+export class InfoboxComponent implements OnInit {
+  public title: string;
+  public description: string;
+  query$: any;
+  resultsearch = '/search';
+  constructor(private knowledgeservice: KnowledgeapiService,
+              private route: Router,
+              private activatedroute: ActivatedRoute,
+              private store: Store<fromRoot.State>,
+              private ref: ChangeDetectorRef) {
+    this.query$ = store.select(fromRoot.getquery);
+    this.query$.subscribe( query => {
+      if (query) {
+        this.knowledgeservice.getsearchresults(query).subscribe(res => {
+          const pageId = Object.keys(res)[0];
+          if (res[pageId].extract) {
+            this.title = res[pageId].title;
+            this.description = res[pageId].extract;
+          } else {
+            this.title = '';
+            this.description = '';
+          }
+        });
+      }
+    });
+  }
+
+  ngOnInit() {
+  }
+
```

Comment:
You did not unsubscribe to the query?

Reply:
In case of no results from wikipedia, I just put description and title to be empty and in html file just give a check that whether description is empty or not using `*ngIf`. Something like 
`<div *ngIf='this.description'>
//code_for_infobox
<div>`
I actually planned for this but I forgot to give a check in `.ts` file.

Reply:
Actually main motive is that you should unsubscribe to the Observable always. It is not affecting anything at this point here, but you should follow this.